docs/includes/fundamentals/as-avs/AtlasSearchTest.php

```diff
@@ -0,0 +1,158 @@
+use MongoDB\Builder\Query;
+use MongoDB\Builder\Search;
+use MongoDB\Driver\Exception\ServerException;
+use MongoDB\Laravel\Schema\Builder;
+use MongoDB\Laravel\Tests\TestCase;
+
+use function array_map;
+use function mt_getrandmax;
+use function rand;
+use function range;
+use function srand;
+use function usleep;
+
+class AtlasSearchTest extends TestCase
+{
+    private array $vectors;
+
+    protected function setUp(): void
+    {
+        require_once __DIR__ . '/Movie.php';
+
+        parent::setUp();
+
+        $moviesCollection = DB::connection('mongodb')->getCollection('movies');
+        $moviesCollection->drop();
+
+        Movie::insert([
+            ['title' => 'Dreaming of Jakarta', 'year' => 1990],
+            ['title' => 'See You in My Dreams', 'year' => 1996],
+            ['title' => 'On the Run', 'year' => 2004],
+            ['title' => 'Jakob the Liar', 'year' => 1999],
+            ['title' => 'Emily Calling Jake', 'year' => 2001],
+        ]);
+
+        Movie::insert($this->addVector([
+            ['title' => 'A', 'plot' => 'A shy teenager discovers confidence and new friendships during a transformative summer camp experience.'],
+            ['title' => 'B', 'plot' => 'A detective teams up with a hacker to unravel a global conspiracy threatening personal freedoms.'],
+            ['title' => 'C', 'plot' => 'High school friends navigate love, identity, and unexpected challenges before graduating together.'],
+            ['title' => 'D', 'plot' => 'Stranded on a distant planet, astronauts must repair their ship before supplies run out.'],
+        ]));
+
+        $moviesCollection = DB::connection('mongodb')->getCollection('movies');
+
+        try {
+            $moviesCollection->createSearchIndex([
+                'mappings' => [
+                    'fields' => [
+                        'title' => [
+                            ['type' => 'string', 'analyzer' => 'lucene.english'],
+                            ['type' => 'autocomplete', 'analyzer' => 'lucene.english'],
+                            ['type' => 'token'],
+                        ],
+                    ],
+                ],
+            ]);
+
+            $moviesCollection->createSearchIndex([
+                'mappings' => ['dynamic' => true],
+            ], ['name' => 'dynamic_search']);
+
+            $moviesCollection->createSearchIndex([
+                'fields' => [
+                    ['type' => 'vector', 'numDimensions' => 4, 'path' => 'vector4', 'similarity' => 'cosine'],
+                    ['type' => 'filter', 'path' => 'title'],
+                ],
+            ], ['name' => 'vector', 'type' => 'vectorSearch']);
+        } catch (ServerException $e) {
+            if (Builder::isAtlasSearchNotSupportedException($e)) {
+                self::markTestSkipped('Atlas Search not supported. ' . $e->getMessage());
+            }
+
+            throw $e;
+        }
+
+        // Wait for the index to be ready
```

Comment:
```suggestion
        // Waits for the index to be ready
```
S: Someone on the team once referenced this doc about code comment consistency in one of my reviews. Since this is more of a descriptive comment, it shouldn't be written as a command. Here is a link to the guidance:

https://docs.google.com/document/d/1Bnhh1OsGRUuXo_6yVUASGWiSX6sWuqrEFxPP6sLZDHM/edit?tab=t.0#heading=h.xv6glj2lnzjh